onnxruntime/core/providers/cuda/reduction/reduction_functions.cu

```diff
@@ -315,22 +306,22 @@ __global__ void reduce_matrix_rows_kernel(const TIn *input, TOut *output, int m,
 
   // Shape is blockDim.y-by-blockDim.x and element type is TBuf.
   extern __shared__ unsigned char shared_memory_[];
-  TBuf *shared_memory = reinterpret_cast<TBuf*>(shared_memory_);
+  TBuf* shared_memory = reinterpret_cast<TBuf*>(shared_memory_);
 
   for (int col = tid_x_in_grid; col < n; col += x_grid_stride) {
     shared_memory[tid_in_block] = TBuf(0.0f);
 
     // This loops load multiple blockDim.y-by-blockDim.x sub-tensors from the input.
     for (int row = tid_y_in_grid; row < m; row += y_grid_stride) {
       TBuf sum = 0.0f;
-      // Thread-level reduction. Each thread loads y_load_count_per_thread values
-      // and aggregrate them.
+// Thread-level reduction. Each thread loads y_load_count_per_thread values
+// and aggregrate them.
 #pragma unroll(y_load_count_per_thread)
       for (int row_inner = 0; row_inner < y_load_count_per_thread; ++row_inner) {
         int row_final = row + row_inner * t_count_y_in_grid;
         int col_final = col;
         if (row_final < m && col_final < n) {
-          sum += TBuf(input[row_final * n + col_final]);
+          sum += TBuf(input[row_final * static_cast<int64_t>(n) + col_final]);
```

Comment:
maybe cast it outside of the loop, this will be performed each time it iterates.

Reply:
done, thanks!